src/findPlugins.js

```diff
@@ -9,13 +9,29 @@ const flatten = require('lodash').flatten;
  * @param  {String} dependency Name of the dependency
  * @return {Boolean}           If dependency is a rnpm plugin
  */
-const isPlugin = (dependency) => !!~dependency.indexOf('rnpm-plugin-');
+const isRNPMPlugin = (dependency) => !!~dependency.indexOf('rnpm-plugin-')
+const isReactNativePlugin = (dependency) => !!~dependency.indexOf('react-native-');
 
-const findPluginInFolder = (folder) => {
-  var pjson;
+const readPackage = (folder) => {
   try {
-    pjson = require(path.join(folder, 'package.json'));
+    return require(path.join(folder, 'package.json'));
   } catch (e) {
+    return null;
```

Comment:
It's a new function, diff might be misleading. Previously we were reading package.json once and returning an array in case nothing was found. This function now returns `package.json` if present.